Add the list of sidebar default property vales with their explanations (#468)
octobercms/docs: plugin-registration.md

**name** | the plugin name, required.
**description** | the plugin description, required.
**author** | the plugin author name, required.

> **icon** | a name of the plugin icon. The full list of available icons can be found in the [UI documentation](../ui/icon). Any icon names provided by this font are valid, for example **icon-glass**, **icon-music**, required.

Reviewer: Is this value actually required? It's only used in one small section in the backend, the plugin details page.

Author: will change it to optional.

Author: https://github.com/octobercms/docs/pull/468/commits/857e97dad12bd2b6b0b259cec5f946801d4b868f

> **iconSvg** | a SVG icon to be used in place of the standard icon. The SVG icon should be a rectangle and can support colors, optional.

Reviewer 2: ```suggestion
**iconSvg** | an SVG icon to be used in place of the standard icon. The SVG icon should be a rectangle and can support colors, optional.
```

To make the sub-menu items visible, you may [set the navigation context](../backend/controllers-ajax#navigation-context) in the back-end controller using the `BackendMenu::setContext` method. This will make the parent menu item active and display the children in the side menu.

Key | Description
------------- | -------------

> **code** | a string value that acts as an unique identifier for that menu option, required.

Reviewer: While this is a property on the MenuItem class, I don't think you can specify it in the array configuration, since it's the key of the configuration.

Author: @LukeTowers 

You've marked both the `code` properties and say they don't come through as an array. But when we print out the following api `BackendMenu::listSideMenuItems();` we see them in the array and get the following result:

```php
Array
(
    [example] => Backend\Classes\SideMenuItem Object
    (
        [code] => example
        [owner] => Acme.Plugin
        [label] => acme.plugin::lang.menu.example.title
        [icon] => fas fa-users-cog
        [iconSvg] =>
        [url] => https://www.example.com/acme/plugin/example
        [counter] =>
        [counterLabel] =>
        [badge] =>
        [order] => 30
        [attributes] => Array
    (
    )
        [permissions] => Array
        (
            [0] => acme.plugin.example
        )
    )
)
```

I thought the doc's should just explain all the properties in the array.

Please advise what to do to resolve your comments.

Author: I see what's confused me, when you output the array `BackendMenu::listSideMenuItems();` it shows `code` in the array. But when people write the php code for the sidebar menu in the `Pluygins.php` file , you don't need to write the `'code' => 'example'` so I will remove both of them.

Reviewer: It can still be documented here, it just needs to have a disclaimer on where a developer would provide that value.

Author: I see, so I'm writing a commit like this:

```
**code** | **this item is auto generated by the cms and should not be added by the developer**, a string value that acts as an unique identifier for that menu option.
**owner** | **this item is auto generated by the cms and should not be added by the developer**, a string value that specifies the menu items owner plugin or module in the format "Author.Plugin".
```

> **owner** | a string value that specifies the menu items owner plugin or module in the format Author, optional.

Reviewer 2: ```suggestion
**owner** | a string value that specifies the menu items owner plugin or module in the format "Author.Plugin", optional.
```

> **owner** | a string value that specifies the menu items owner plugin or module in the format "Author.Plugin", optional.

Reviewer: This also can't be specified in the array configuration AFAIK

Author: I see what's confused me, when you output the array `BackendMenu::listSideMenuItems();` it shows `code` but when people write the php code for the sidebar menu, you don't need to write the `'code' => 'example'` so I will remove both of them.

Author: I see what's confused me, when you output the array `BackendMenu::listSideMenuItems();` it shows `code` in the array. But when people write the php code for the sidebar menu in the `Pluygins.php` file , you don't need to write the `'owner' => 'example'` so I will remove both of them.

Reviewer: It can still be documented here, it just needs to have a disclaimer on where a developer would provide that value.

Author: I see, so I'm writing a commit like this:

```
**code** | **this item is auto generated by the cms and should not be added by the developer**, a string value that acts as an unique identifier for that menu option.
**owner** | **this item is auto generated by the cms and should not be added by the developer**, a string value that specifies the menu items owner plugin or module in the format "Author.Plugin".
```

**label** | specifies the menu label localization string key, required.

> **icon** | an icon name from the Font Awesome icon collection, required.

Reviewer 2: ```suggestion
**icon** | an icon name from the [October CMS icon collection](../ui/icon), required.
```

**iconSvg** | an SVG icon to be used in place of the standard icon, the SVG icon should be a rectangle and can support colors, optional.

> **url** | the back-end relative URL the menu item should point to, required.

Reviewer: ```suggestion
**url** | the URL the menu item should point to (ex. `Backend::url('author/plugin/controller/action')`, required.
```

**counter** | a numeric value to output near the menu icon. The value should be a number or a callable returning a number, optional.
**counterLabel** | a string value to describe the numeric reference in counter, optional.

> **badge** | a string value to output near the menu icon, the value should be a string, this value will override the counter if set, optional.

Reviewer: ```suggestion
**badge** | a string value to output in place of the counter, the value should be a string and will override the badge property if set, optional.
```

> **attributes** | an array of attributes and values to apply to the menu item, optional.

Reviewer: ```suggestion
**attributes** | an associative array of attributes and values to apply to the menu item, optional.
```

> **permissions** | an array of permissions the back-end user should have, optional.

Reviewer: ```suggestion
**permissions** | an array of permissions the backend user must have in order to view the menu item (Note: direct access of URLs still requires separate permission checks), optional.
```

> **code** | **this item is auto generated by the cms and should not be added by the developer**, a string value that acts as an unique identifier for that menu option.

Reviewer: ```suggestion
**code** | a string value that acts as an unique identifier for that menu option. **NOTE**: This is a system generated value and should not be provided when registering the navigation items.
```

> **owner** | **this item is auto generated by the cms and should not be added by the developer**, a string value that specifies the menu items owner plugin or module in the format "Author.Plugin".

Reviewer: ```suggestion
**owner** | a string value that specifies the menu items owner plugin or module in the format "Author.Plugin". **NOTE**: This is a system generated value and should not be provided when registering the navigation items.
```

Key | Description | Required field
------------- | ------------- | -------------
**code** | a string value that acts as an unique identifier for that menu option. | **required**
**owner** | a string value that specifies the menu items owner plugin or module in the format Author. | optional
**label** | specifies the menu label localization string key. | **required**
**icon** | an icon name from the Font Awesome icon collection. | **required**
**iconSvg** | an SVG icon to be used in place of the standard icon, the SVG icon should be a rectangle and can support colors. | optional
**url** | the back-end relative URL the menu item should point to. | **required**
**counter** | a numeric value to output near the menu icon. The value should be a number or a callable returning a number. | optional
**counterLabel** | a string value to describe the numeric reference in counter. | optional
**badge** | a string value to output near the menu icon, the value should be a string, this value will override the counter if set. | optional
**attributes** | an array of attributes and values to apply to the menu item. | optional

> **group** | a string value that defines a sub-header to a group of menu fields. | optional

Reviewer: I don't believe this is actually an officially supported property.
